core/domain/exp_domain_test.py

```diff
@@ -344,6 +344,40 @@ def test_validation(self):
                 utils.ValidationError, 'destination ABC is not a valid'):
             exploration.validate()
 
+        # Ensure an answer group with two fuzzy rules is invalid
+        init_state = exploration.states[exploration.init_state_name]
+        default_outcome = init_state.interaction.default_outcome
+        default_outcome.dest = exploration.init_state_name
```

Comment:
Unless I'm mistaken, isn't ``default_outcome.dest = exploration.init_state_name`` restoring the exploration to a valid state?

If that's right, I'm sort of suggesting this as a general structure:

```
        # Test whether a default outcome to a non-existing state is invalid.
        exploration.states = {exploration.init_state_name: new_state}
        with self.assertRaisesRegexp(
                utils.ValidationError, 'destination ABC is not a valid'):
            exploration.validate()

        # restore exploration
        exploration.validate()

        # Ensure an answer group with two fuzzy rules is invalid
        ...
```

If this isn't super clear, then please ping me so we can sort it out in more detail. Thanks!